frontend/src/components/Chat/Chat.jsx

```diff
@@ -0,0 +1,123 @@
+import "./Chat.css";
+import { useState, useEffect, useContext } from "react";
+import ScrollToBottom from "react-scroll-to-bottom";
+import axios from "axios";
+import { Context } from "../../context";
+import { formatDateTime } from "../../utils";
+
+export default function Chat({ socket, user, room, friend }) {
+    const [currentMessage, setCurrentMessage] = useState("");
+    const [messageList, setMessageList] = useState([]);
+    const { setErrorMessage } = useContext(Context);
+
+    useEffect(() => {
+        const fetchMessages = async () => {
+            try {
+                const response = await axios.get(
+                    `${import.meta.env.VITE_HOST}/messages/${friend.user2.id}`,
+                    { withCredentials: true, validateStatus: () => true }
+                );
+    
+                if (response.status === 200) {
+                    setMessageList(response.data.messages);
+                }
+
+                if (response.status === 500) {
+                    setErrorMessage(response.data.error);
+                }
+            } catch (error) {
+                setErrorMessage(`${error.message}: Please try again later.`);
+            }
+            const response = await axios.get(
+                `${import.meta.env.VITE_HOST}/messages/${friend.user2.id}`,
```

Comment:
Why is this call repeated?

Reply:
That call is to get the chat history with your friend. The call in ChatRoom just gets a list of all the friends you have.